### PR TITLE
[JENKINS-67823] Don't append leading slash to foreign icons

### PR DESCRIPTION
**Pull request by NotMyFault**
Fixes [JENKINS-67823](https://issues.jenkins.io/browse/JENKINS-67823)

The change proposed removes the leading slash for foreign icons, either added by other plugins or read from URLs, like the GitLab plugin does.

### Proposed changelog entries

* Don't add leading slash to foreign icons (regression in 2.335).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by MarkEWaite** [comment]
I built the change and confirmed that it resolves the issue I reported in [JENKINS-67823](https://issues.jenkins.io/browse/JENKINS-67823)

**Review by MarkEWaite — APPROVED** [review]

File: test/src/test/java/lib/layout/IconTest.java
Comment (MarkEWaite) [inline test/src/test/java/lib/layout/IconTest.java:154]: Based on the assertion failure, I think modifying the test is the wrong direction.

The assertion failure says:

```
Expected: a string ending with "/plugin/xxx/icon.png"
     but: was "/jenkinsplugin/xxx/icon.png"
```

If the Jenkins controller is configured to run with the prefix "/jenkins" (behind an nginx reverse proxy like mine is), then the value currently returned ("/jenkinsplugin/xxx/icon.png") seems incorrect.  I think it should be returning "/jenkins/plugin/xxx/icon.png".

Unfortunately, I don't know the cases where it should not be adding the leading slash.  Maybe the conditional that was deleted needs to be replaced with a check if the string starts with "image" or "plugin" and only then add the leading slash?

Comment (NotMyFault) [reply test/src/test/java/lib/layout/IconTest.java]: > Based on the assertion failure, I think modifying the test is the wrong direction.

Yeah, I was quickly messing with things. Already reverted 👍🏻 

Functions#tryGetIconPath determines the icon source by either going off /images/ or /plugin/, but that check doesn't suffice for plugins which add the icon from an URL.
I've added an exemption for http, that suffices for plugins which read icons from foreign sources.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
